lib/src/decoder.dart

```diff
@@ -83,8 +83,7 @@ class Decoder extends Converter<String, Map<String, dynamic>> {
 
   bool _isListKey(String key) {
     try {
-      int.parse(key);
-      return true;
+      return int.tryParse(key)!=null;
```

Comment:
Hi @williamyeung72, can you add spacing like the `dart format` would format this ?

`int.tryParse(key) != null;`

And can you add a test to avoid future regression on this feature ?